### PR TITLE
 Measuring throughput of connectors 

### LINKED ISSUE #2524: Measuring throughput of connectors
## Tell us about the problem you're trying to solve
We have no/little visibility on the throughput of our connectors. 

## Describe the solution you’d like
Add properties duration / volume_rows / volume_MB for Connector Jobs when job_type=SYNC. 

Here's the updated Segment spreadsheet: https://docs.google.com/spreadsheets/d/1qQPMBdz86yMBiyXpjw91D3HoP6hN7yIPRmLrF7xWOgA/edit#gid=1567609168

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Closes #2524
 
## How
Add attempts metadata when tracking job completions

## Recommended reading order
1. `airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-appsflyer-singer/unit_tests/unit_test.py [hunk 1]
```diff
@@ -22,5 +22,6 @@
 SOFTWARE.
 """
 
+
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-appsflyer-singer/unit_tests/unit_test.py:25]: blankspace.

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java [hunk 2]
```diff
@@ -204,6 +211,25 @@ public void trackSync(Job job, JobState jobState) {
...
+    final Builder<String, Object> metadata = ImmutableMap.builder();
+    final Job job = jobPersistence.getJob(jobId);
+    if (jobState != JobState.STARTED && job != null) {
+      final List<Attempt> attempts = job.getAttempts();
+      if (attempts != null && !attempts.isEmpty()) {
+        final Attempt lastAttempt = attempts.get(attempts.size() - 1);
+        if (lastAttempt.getOutput() != null && lastAttempt.getOutput().isPresent()) {
+          final JobOutput jobOutput = lastAttempt.getOutput().get();
+          final StandardSyncSummary syncSummary = jobOutput.getSync().getStandardSyncSummary();
+          metadata.put("duration", syncSummary.getEndTime() - syncSummary.getStartTime());
```
Comment (cgardens) [inline airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java:224]: just want to sanity check that sync summary metrics are in seconds?

**Review by cgardens — APPROVED** [review]
lgtm. 

worth noting that fetching the job output struck me that it could be dangerous because i didn't know what order saving the job output and doing the track calls happened. it looks like we do save job output before calling track so it should be okay.

File: airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java (hunk 2 above)
Comment (ChristopheDuong) [reply airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java:224]: it's in milliseconds based on what's being saved in the syncSummary:
https://github.com/airbytehq/airbyte/blob/0c603db2966ee35b7ed9b62e5c95947522cab429/airbyte-workers/src/main/java/io/airbyte/workers/DefaultSyncWorker.java#L84

Comment (cgardens) [reply airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java:224]: if you look at the excel spreadsheet that john put together for the tracking plan he is expecting seconds. either his expectation needs to change to ms or we need to convert here. i have no preference, just want to make sure that the two of you are speaking the same language.

Comment (ChristopheDuong) [reply airbyte-scheduler/src/main/java/io/airbyte/scheduler/JobTracker.java:224]: Ok will switch to seconds, thanks!

**Pull request merged**